pkg/upload/js.go

```diff
@@ -16,9 +24,163 @@ type JsOptions struct {
 	Path        utils.Paths `arg:"" name:"path" help:"Path to a directory of source maps and bundles to upload" type:"path" default:"."`
 }
 
+// Resolve the project if it isn't specified using the currend working directory
+func resolveProjectRoot(projectRoot string, path string) string {
+	if projectRoot != "" {
+		return projectRoot
+	}
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		return path
+	}
+	return workingDirectory
+}
+
+// Attempt to parse information from the package.json file if values aren't provided on the command line
+func ResolveVersion(versionName string, projectRoot string, logger log.Logger) (string, error) {
+	if versionName != "" {
+		return versionName, nil
+	}
+	checkPath, err := filepath.Abs(projectRoot)
+	if err != nil {
+		return "", fmt.Errorf("unable to make project root an absolute path %s", projectRoot)
+	}
+	// Walk up the folder structure as far as possible
+	for filepath.Dir(checkPath) != checkPath {
+		packageJson := filepath.Join(checkPath, "package.json")
+		if !utils.FileExists(packageJson) {
+			checkPath = filepath.Dir(checkPath)
+			continue
+		}
+		file, err := os.Open(packageJson)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to open %s: %s", packageJson, err))
+			return "", nil
+		}
+		byteValue, err := io.ReadAll(file)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to read %s: %s", packageJson, err))
+			return "", nil
+		}
+		var parsedPackageJson map[string]interface{}
+		err = json.Unmarshal(byteValue, &parsedPackageJson)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to parse %s: %s", packageJson, err))
+			return "", nil
+		}
+		if parsedPackageJson["version"] == nil {
+			logger.Warn(fmt.Sprintf("missing version in %s", packageJson))
+			return "", nil
+		}
+		appVersion := parsedPackageJson["version"].(string)
+		logger.Info(fmt.Sprintf("Using app version from %s: %s", packageJson, appVersion))
+		return appVersion, nil
+	}
+	return "", nil
+}
+
+// Attempt to find the source maps by walking the build directory if it is not passed in to the command line
+func ResolveSourceMapPaths(sourceMap string, outputPath string, projectRoot string, logger log.Logger) ([]string, error) {
+	if sourceMap != "" {
+		sourceMap := sourceMap
+		if !path.IsAbs(sourceMap) {
```

Comment:
As discussed IRL, let's not prepend project root and assume that the path provided is relative to the running app.